shell/renderer/web_worker_observer.cc

```diff
@@ -50,6 +55,18 @@ WebWorkerObserver::~WebWorkerObserver() {
 
 void WebWorkerObserver::WorkerScriptReadyForEvaluation(
     v8::Local<v8::Context> worker_context) {
+  auto* execution_context = blink::ExecutionContext::From(worker_context);
+
+  // We do not create a Node.js environment in service or shared workers
+  // owing to an inability to customize sandbox policies in these workers
+  // given that they're run out-of-process.
+  bool is_service_worker = execution_context->IsServiceWorkerGlobalScope();
+  bool is_shared_worker = execution_context->IsSharedWorkerGlobalScope();
+  if (is_service_worker || is_shared_worker) {
```

Comment:
What about moving this check one layer above into `ElectronRendererClient::WorkerScriptReadyForEvaluationOnWorkerThread` so that it can be inlined and avoid creating this class when not needed.